[MSVC] Terminal failed to build with msvc due to error C2664 on Windows

### Windows Terminal version

_No response_

### Windows build number

10.0.20348.0

### Other Software

_No response_

### Steps to reproduce


```cmd
git clone https://github.com/microsoft/termina F:\repo\microsoft\terminal
git -C "F:\repo\microsoft\terminal" reset --hard 34aa6aa
git -C "F:\repo\microsoft\terminal" submodule sync
git -C "F:\repo\microsoft\terminal" submodule foreach git reset --hard
git -C "F:\repo\microsoft\terminal" submodule update --init --recursive

cd F:\repo\microsoft\terminal
set path=F:\tools\Nuget\6.5.0;%PATH%
nuget restore OpenConsole.sln 2>&1
nuget restore dep\nuget\packages.config

set VSCMD_SKIP_SENDTELEMETRY=1 & "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\Tools\VsDevCmd.bat" -host_arch=x86 -arch=x86
set _CL_=/wd4189 /wd5104 /D_SILENCE_STDEXT_HASH_DEPRECATION_WARNINGS %_CL_%
cd F:\repo\microsoft\terminal
msbuild /m /p:Platform=x86 /p:Configuration=Release /p:WindowsTargetPlatformVersion=10.0.22621.0 OpenConsole.sln /t:Restore /t:Rebuild 2>&1
```

I tried the latest commit: 10c6206, the issue still can be reproduced.

### Expected Behavior

Build successfully.

### Actual Behavior

error info:
```cmd
44>F:\repo\microsoft\terminal\src\cascadia\TerminalControl\TermControl.cpp(859,27): error C2664: 'winrt::hstring &winrt::hstring::operator =(winrt::hstring &&)': cannot convert argument 1 from 'initializer list' to 'winrt::hstring &&' [F:\repo\microsoft\terminal\src\cascadia\TerminalControl\TerminalControlLib.vcxproj]
   F:\repo\microsoft\terminal\src\cascadia\TerminalControl\TermControl.cpp(859,27): message : copy-list-initialization of 'winrt::hstring' cannot use an explicit constructor [F:\repo\microsoft\terminal\src\cascadia\TerminalControl\TerminalControlLib.vcxproj]
   F:\repo\microsoft\terminal\src\cascadia\TerminalControl\Generated Files\winrt\base.h(2889,9): message : see declaration of 'winrt::hstring::hstring' (compiling source file TermControl.cpp) [F:\repo\microsoft\terminal\src\cascadia\TerminalControl\TerminalControlLib.vcxproj]
44>F:\repo\microsoft\terminal\src\cascadia\TerminalControl\TermControl.cpp(864,27): error C2664: 'winrt::hstring &winrt::hstring::operator =(winrt::hstring &&)': cannot convert argument 1 from 'initializer list' to 'winrt::hstring &&' [F:\repo\microsoft\terminal\src\cascadia\TerminalControl\TerminalControlLib.vcxproj]
   F:\repo\microsoft\terminal\src\cascadia\TerminalControl\TermControl.cpp(864,27): message : copy-list-initialization of 'winrt::hstring' cannot use an explicit constructor [F:\repo\microsoft\terminal\src\cascadia\TerminalControl\TerminalControlLib.vcxproj]
   F:\repo\microsoft\terminal\src\cascadia\TerminalControl\Generated Files\winrt\base.h(2889,9): message : see declaration of 'winrt::hstring::hstring' (compiling source file TermControl.cpp) [F:\repo\microsoft\terminal\src\cascadia\TerminalControl\TerminalControlLib.vcxproj]
44>F:\repo\microsoft\terminal\src\cascadia\TerminalControl\TermControl.cpp(868,27): error C2664: 'winrt::hstring &winrt::hstring::operator =(winrt::hstring &&)': cannot convert argument 1 from 'initializer list' to 'winrt::hstring &&' [F:\repo\microsoft\terminal\src\cascadia\TerminalControl\TerminalControlLib.vcxproj]
   F:\repo\microsoft\terminal\src\cascadia\TerminalControl\TermControl.cpp(868,27): message : copy-list-initialization of 'winrt::hstring' cannot use an explicit constructor [F:\repo\microsoft\terminal\src\cascadia\TerminalControl\TerminalControlLib.vcxproj]
   F:\repo\microsoft\terminal\src\cascadia\TerminalControl\Generated Files\winrt\base.h(2889,9): message : see declaration of 'winrt::hstring::hstring' (compiling source file TermControl.cpp) [F:\repo\microsoft\terminal\src\cascadia\TerminalControl\TerminalControlLib.vcxproj]
```

--- a/src/cascadia/TerminalApp/TabManagement.cpp
+++ b/src/cascadia/TerminalApp/TabManagement.cpp
@@ -434,7 +434,7 @@ namespace winrt::TerminalApp::implementation
                     // environment variables, but the user might have set one in
                     // the settings. Expand those here.
 
-                    path = { wil::ExpandEnvironmentStringsW<std::wstring>(path.c_str()) };
+                    path = winrt::hstring{ wil::ExpandEnvironmentStringsW<std::wstring>(path.c_str()) };
                 }
 
                 if (!path.empty())

--- a/src/cascadia/TerminalControl/TermControl.cpp
+++ b/src/cascadia/TerminalControl/TermControl.cpp
@@ -907,7 +907,7 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                 HRESULT_FROM_WIN32(ERROR_PATH_NOT_FOUND) == hr)
             {
                 message = winrt::hstring{ fmt::format(std::wstring_view{ RS_(L"PixelShaderNotFound") },
-                                        (_focused ? _core.FocusedAppearance() : _core.UnfocusedAppearance()).PixelShaderPath()) };
+                                                      (_focused ? _core.FocusedAppearance() : _core.UnfocusedAppearance()).PixelShaderPath()) };
             }
             else if (D2DERR_SHADER_COMPILE_FAILED == hr)
             {
@@ -916,7 +916,7 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             else
             {
                 message = winrt::hstring{ fmt::format(std::wstring_view{ RS_(L"UnexpectedRendererError") },
-                                        hr) };
+                                                      hr) };
             }
 
             auto noticeArgs = winrt::make<NoticeEventArgs>(NoticeLevel::Warning, std::move(message));

--- a/src/cascadia/TerminalControl/TermControl.cpp
+++ b/src/cascadia/TerminalControl/TermControl.cpp
@@ -906,16 +906,16 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             if (HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND) == hr ||
                 HRESULT_FROM_WIN32(ERROR_PATH_NOT_FOUND) == hr)
             {
-                message = { fmt::format(std::wstring_view{ RS_(L"PixelShaderNotFound") },
+                message = winrt::hstring{ fmt::format(std::wstring_view{ RS_(L"PixelShaderNotFound") },
                                         (_focused ? _core.FocusedAppearance() : _core.UnfocusedAppearance()).PixelShaderPath()) };
             }
             else if (D2DERR_SHADER_COMPILE_FAILED == hr)
             {
-                message = { fmt::format(std::wstring_view{ RS_(L"PixelShaderCompileFailed") }) };
+                message = winrt::hstring{ fmt::format(std::wstring_view{ RS_(L"PixelShaderCompileFailed") }) };
             }
             else
             {
-                message = { fmt::format(std::wstring_view{ RS_(L"UnexpectedRendererError") },
+                message = winrt::hstring{ fmt::format(std::wstring_view{ RS_(L"UnexpectedRendererError") },
                                         hr) };
             }
 

